bloom-housing/bloom
# feat: switching to write stream and sequential building

This PR addresses https://github.com/metrotranscom/doorway/issues/753
and also addresses: https://github.com/bloom-housing/bloom/issues/4337 

blocked by https://github.com/bloom-housing/bloom/pull/4322


## Description
This switches the way we write the spreadsheet to leveraging a write stream, and also changes from a promise.All() call to handle the promises to instead handle that sequentially

## How Can This Be Tested/Reviewed?
go in and export a large lottery, you should notice the time required to build the export has decreased, and if you are tracking the mem usage that should drop significantly from what it was previously 

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
